### PR TITLE
Metadata entries for early failures. Closes #966

### LINKED ISSUE #966: Add metadata for WorkflowMaterialiserActor Failure
If there is a failure in the `WorkflowMaterialiserActor` phase, there is no external way to debug the error.

Allow the `WorkflowMaterialiserActor` to send failure messages to the metadata service in this instance, so that it's possible to debug errors!

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Comment by mcovarr** [comment]
:+1:

File: engine/src/test/scala/cromwell/engine/backend/WorkflowDescriptorSpec.scala [hunk 1]
```diff
@@ -53,7 +53,8 @@ trait WorkflowDescriptorBuilder {
     implicit val timeout = akka.util.Timeout(awaitTimeout)
     implicit val ec = actorSystem.dispatcher
 
-    val actor = actorSystem.actorOf(MaterializeWorkflowDescriptorActor.props(), "MaterializeWorkflowDescriptorActor-" + id.id)
+    val serviceRegistryIgnorer = actorSystem.actorOf(DevNullActor.props)
```
Comment (geoffjentry) [inline engine/src/test/scala/cromwell/engine/backend/WorkflowDescriptorSpec.scala]: I was super scared that we had `DevNullActor` until I noticed the name of this file

Comment (kshakir) [inline engine/src/test/scala/cromwell/engine/backend/WorkflowDescriptorSpec.scala]: [`Props.empty`](https://github.com/akka/akka/blob/v2.3.12/akka-actor/src/main/scala/akka/actor/Props.scala#L49) does the same as your `DevNullActor.props`:

```scala
val serviceRegistryIgnorer = actorSystem.actorOf(Props.empty)
```

Comment (Horneth) [inline engine/src/test/scala/cromwell/engine/backend/WorkflowDescriptorSpec.scala]: You can also just do `val serviceRegistryIgnorer = TestProbe().ref`

**Comment by scottfrazer** [comment]
👍

**Pull request merged**